fortnite_api/cosmetics/instrument.py

```diff
@@ -0,0 +1,103 @@
+"""
+MIT License
+
+Copyright (c) 2019-present Luc1412
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from __future__ import annotations
+
+import datetime
+from typing import Any, Dict, List, Optional, Tuple
+
+from ..http import HTTPClientT
+from ..utils import get_with_fallback, parse_time
+from .common import Cosmetic, CosmeticImages, CosmeticRarity, CosmeticSeries, CosmeticType
+
+__all__: Tuple[str, ...] = ('CosmeticInstrument',)
+
+
+class CosmeticInstrument(Cosmetic[HTTPClientT]):
+    """Represents an instrument cosmetic in Fortnite.
+
+    This class inherits from :class:`Cosmetic`.
+
+    Attributes
+    ----------
+    name: :class:`str`
+        The name of the instrument.
+    description: :class:`str`
+        The description of the instrument.
+    type: Optional[:class:`CosmeticType`]
+        The type of the instrument.
+    rarity: Optional[:class:`CosmeticRarity`]
+        The rarity of the instrument.
+    images: Optional[:class:`CosmeticImages`]
+        Any instrument images.
+    series: Optional[:class:`CosmeticSeries`]
+        The series of the instrument.
+    gameplay_tags: List[:class:`str`]
+        The gameplay tags of the instrument.
+    path: Optional[:class:`str`]
+        The path of the instrument.
+    showcase_video: Optional[:class:`str`]
+        The showcase video of the instrument.
```

Comment:
Since it's a Youtube Video ID, we should be more precise. We could rename it to `showcase_video_id` and mention in the docs it's a YoutTube URL or to `showcase_video_url` and provide the full youtube url.